services/ui-src/src/utils/s3Uploader.js

```diff
@@ -68,27 +68,38 @@ export async function uploadFile(file) {
   const fileToUpload = ensureLowerCaseFileExtension(file);
 
   let retPromise;
+  let numTries = 1;
   const targetPathname = `${Date.now()}/${fileToUpload.name}`;
 
-  try {
-    const stored = await Storage.vault.put(targetPathname, fileToUpload, {
-      level: "protected",
-      contentType: fileToUpload.type,
-    });
+  while (numTries > 0 && numTries < 10) {
+    try {
+      const stored = await Storage.put(targetPathname, fileToUpload, {
+        level: "protected",
+        contentType: fileToUpload.type,
+      });
 
-    const url = await Storage.vault.get(stored.key, { level: "protected" });
+      const url = await Storage.get(stored.key, { level: "protected" });
 
-    let result = {
-      s3Key: stored.key,
-      filename: fileToUpload.name,
-      contentType: fileToUpload.type,
-      url: url.split("?", 1)[0], //We only need the permalink part of the URL since the S3 bucket policy allows for public read
-      title: fileToUpload.title,
-    };
+      let result = {
+        s3Key: stored.key,
+        filename: fileToUpload.name,
+        contentType: fileToUpload.type,
+        url: url.split("?", 1)[0], //We only need the permalink part of the URL since the S3 bucket policy allows for public read
+        title: fileToUpload.title,
+      };
 
-    retPromise = Promise.resolve(result);
-  } catch (error) {
-    retPromise = Promise.reject(error);
+      retPromise = Promise.resolve(result);
+      numTries = 0;
+    } catch (error) {
+      numTries++;
+      retPromise = Promise.reject(error);
+      if (error.message.indexOf("failed with status code 503") !== -1) {
+        console.log("got 503, numTries: ", numTries);
+      } else {
+        console.log("got a different error: ", error);
```

Comment:
doesnt affect anything but do we need the logic branch here to detect the 503 if in both cases we are just incrementing the try counter and trying again? In both cases we could just log the error object to console (which would include the errror.message).

Reply:
HA HA - good eye!  That was leftover from when I was not retrying on anything but 503s.... I went through a lot of trial and error on this thing and well, apparently I left a "spiderweb" hanging :)